kotlinx-coroutines-core/jvm/src/Future.kt

```diff
@@ -5,46 +5,26 @@ package kotlinx.coroutines
 
 import java.util.concurrent.*
 
-/**
- * Cancels a specified [future] when this job is cancelled.
- * This is a shortcut for the following code with slightly more efficient implementation (one fewer object created).
- * ```
- * invokeOnCompletion { if (it != null) future.cancel(false) }
- * ```
- *
- * @suppress **This an internal API and should not be used from general code.**
- */
-@InternalCoroutinesApi
-public fun Job.cancelFutureOnCompletion(future: Future<*>): DisposableHandle =
-    invokeOnCompletion(handler = CancelFutureOnCompletion(future))
-
 /**
  * Cancels a specified [future] when this job is cancelled.
  * This is a shortcut for the following code with slightly more efficient implementation (one fewer object created).
  * ```
  * invokeOnCancellation { if (it != null) future.cancel(false) }
  * ```
  */
+@Deprecated(
```

Comment:
Please drop a comment since what version it's deprecated as we do in other places